### PR TITLE
no grav movement now uses !can_pass as well as density

### LINKED ISSUE #4021: no gravity around the blob
shit's bugged yo

### PR DESCRIPTION
**Pull request by MrsTonedOne**
Fixes #4021 no gravity around the blob

:cl:
tweak: Movement in no gravity now allows you to push off of things only dense to some mobs if it is dense to you (such as tables/blob tiles) rather than just objects that are only dense to all mobs.
/:cl:

### CODE REVIEW

File: code/modules/mob/mob_movement.dm [hunk 1]
```diff
@@ -328,7 +328,7 @@
 			var/atom/movable/AM = A
 			if(AM == buckled) //Kind of unnecessary but let's just be sure
 				continue
-			if(AM.density)
+			if(!AM.CanPass(src))
```
Comment (phil235) [inline code/modules/mob/mob_movement.dm]: Won't this mean that monkey can't use tables to move in zero G? And small animals won't be able to use grille.
We could try `if(!AM.CanPass(src) || AM.density)`

Comment (MrStonedOne) [inline code/modules/mob/mob_movement.dm]: point. i guess what would work.

**Comment by Anonus** [comment]
:+1:
It'd be nice if this was separate from CanPass() and density so you could do it arbitrarily, so that, say, blob spores could ignore lack of gravity near blob tiles instead of drifting near blob tiles.

**Comment by duncathan** [comment]
why can't spess men swim anyway

**Pull request merged**